lib/tokenize.cpp

```diff
@@ -9239,6 +9239,21 @@ void Tokenizer::findGarbageCode() const
             syntaxError(tok);
         if (Token::Match(tok, "[+-] [;,)]}]") && !(isCPP() && Token::Match(tok->previous(), "operator [+-] ;")))
             syntaxError(tok);
+        if (Token::simpleMatch(tok, ",")) {
+            if (Token::Match(tok->previous(), "(|[|{|<|%assign%|%or%|%oror%|==|!=|+|-|/|!|>=|<=|~|^|::|sizeof|throw|decltype|typeof"))
+                syntaxError(tok);
```

Comment:
For information.. I want to refactor these somehow so we don't say just "Syntax error". We should write a little more.

Users sometime wonder why Cppcheck thinks there is a syntax error.

Reply:
If Cppcheck is wrong .. I guess it would be easier to see that.